sycl/include/CL/sycl/bit_cast.hpp

```diff
@@ -19,7 +19,13 @@ namespace sycl {
 
 // forward decl
 namespace detail {
-inline void memcpy(void *Dst, const void *Src, std::size_t Size);
+inline void memcpy_bit_cast(void *Dst, const void *Src, std::size_t Size) {
```

Comment:
Am I right that this function is used to make `sycl::bit_cast` `constexpr` if no `__builtin_bit_cast` and `std::bit_cast` are available? If so, then it should probably needs to be marked as `constexpr`

Reply:
Yes, it is right version of function working.  But it uses reinterpret_cast, that can not be used in compile time

Reply:
But unless it is marked `constexpr` it can't be used in `constexpr` context - how it is different from `memcpy` then and why do we need to replace one thing which doesn't work with another thing which doesn't work as well?

Reply:
This memcpy will be used only if std::bit_cast and __builtin_bit_cast are not defined. I defined macro that do the functions of half constexpr or not. So if we don't have opportunity to use std/builtin_bit_cast then half functions will be non-constexpr and will use non-constexpr memcpy.  Did I understand the question correctly?

Reply:
> So if we don't have opportunity to use std/builtin_bit_cast then half functions will be non-constexpr and will use non-constexpr memcpy. Did I understand the question correctly?

Right, it is not clear to me why do we need to introduce an extra lines of code if they do not bring any benefits to us

Reply:
Why was this resolved? I agree with Alexey that it makes no sense to replace one memcpy with another. Given that the new one has no advantage.

Reply:
If I have the only declaration of memcpy in bit_cast.hpp then sycl build fails with error "function memcpy used but never defined". half_type.cpp has no declaration of memcpy. I renamed memcpy for bit_cast.hpp and moved declaration there for fix this problem

Reply:
> If I have the only declaration of memcpy in bit_cast.hpp then sycl build fails with error "function memcpy used but never defined".

So, probably you just need to include one more header where `detail::memcpy` is defined?

Reply:
Added include to half_type.cpp